applications/FluidDynamicsApplication/tests/test_fluid_rve.py

```diff
@@ -82,14 +87,53 @@ def _aux_fluid_rve_computation(self, parameters):
 
                 }
             }]''')
-            output_settings[0]["Parameters"]["output_name"].SetString("FluidRVETest/fluid_rve_test_{}D".format(domain_size))
+            output_settings[0]["Parameters"]["output_name"].SetString("FluidRVETest/fluid_rve_test_{}D".format(self.domain_size))
             parameters["output_processes"].AddValue("gid_output", output_settings)
         
         model = KratosMultiphysics.Model()
-        simulation = FluidDynamicsAnalysisRVE(model, parameters)
-        simulation.Run()
+        self.simulation = FluidDynamicsAnalysisRVE(model, parameters)
+        self.simulation.Run()
+               
+        self._CheckResults()
+        
+    def _CheckResults(self):
+        model_part = self.simulation._GetSolver().GetComputingModelPart()
         
-        # Space for testing
+        if self.print_reference_values:
+            with open('FluidRVETest/' + self.reference_file + '.csv','w') as ref_file:
+                if self.domain_size == 2:
+                    ref_file.write("#ID, VELOCITY_X, VELOCITY_Y\n")
+                else:
+                    ref_file.write("#Id, VELOCITY_X, VELOCITY_Y, VELOCITY_Z\n")
+                for node in model_part.Nodes:
+                    vel = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY,0)
+                    if self.domain_size == 2:
+                        ref_file.write("{0}, {1}, {2}\n".format(node.Id, vel[0], vel[1]))
+                    else:
+                        ref_file.write("{0}, {1}, {2}, {3}\n".format(node.Id, vel[0], vel[1], vel[2]))
+        else:
+            with open('FluidRVETest/' + self.reference_file + '.csv','r') as reference_file:
+                reference_file.readline() # skip header
+                line = reference_file.readline()
+
+                for node in model_part.Nodes:
+                    values = [ float(i) for i in line.rstrip('\n ').split(',') ]
+                    reference_vel_x = values[1]
+                    reference_vel_y = values[2]
+                    if self.domain_size == 3:
+                        reference_vel_z = values[3]
+
+                    velocity = node.GetSolutionStepValue(KratosMultiphysics.VELOCITY)
+                    self.assertAlmostEqual(reference_vel_x, velocity[0], delta = self.check_tolerance)
+                    self.assertAlmostEqual(reference_vel_y, velocity[1], delta = self.check_tolerance)
+                    if self.domain_size == 3:
+                        self.assertAlmostEqual(reference_vel_z, velocity[2], delta = self.check_tolerance)
+                   
+
+                    line = reference_file.readline()
+                if line != '': 
+                    self.fail("The number of nodes in the mdpa is smaller than the number of nodes in the output file")
```

Comment:
Though it works, this is how we used to check results in the past when we didn't have the mechanism to properly do so. To avoid reimplementing this in each test, we created some processes. See how these can be used in here

https://github.com/KratosMultiphysics/Kratos/blob/9e351bc965ef0a8347ccd495eadadad1ce74053e/applications/FluidDynamicsApplication/tests/embedded_couette_flow_test.py#L253-L295

I think this will make things easier.

Reply:
Done.